.github/workflows/continuous-integration.yml

```diff
@@ -2,43 +2,67 @@ name: Create Release
 on: push
 
 jobs:
-  create_release:
-    name: build
-    runs-on: ubuntu-latest
-    if: github.ref == 'refs/heads/master'
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-        with:
-          fetch-depth: 0  # This causes all history to be fetched, which is required for calculate-version to function	
-
-      - name: Install Python 3.8
-        uses: actions/setup-python@v1
-        with:
-          python-version: 3.8
-
-      - name: Upgrade python pip
-        run: python -m pip install --upgrade pip 
-
-      - name: Install git
-        run: pip install gitpython
-
-      - name: Install semver
-        run: pip install semver
-
-      - name: Set SPEC_VERSION env var
-        run: echo ::set-env name=SPEC_VERSION::$(python scripts/calculate_version.py)
-        env:
-          ACTIONS_ALLOW_UNSECURE_COMMANDS: true
-
-      - name: Create release (master only)
-        id: create-release
+    node_linting:
+        name: Linting
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+              with:
+                fetch-depth: 0  # This causes all history to be fetched, which is required for calculate-version to function
+            - name: Perform Setup
+              run: make install-node
+            - name: Run linting
+              run: npm run lint
+    postman_testing:
+        name: Running postman tests
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+              with:
+                fetch-depth: 0  # This causes all history to be fetched, which is required for calculate-version to function
+            - name: Perform Setup
+              run: make install-node
```

Comment:
Are we repeating ourselves here, and could this be done once as a dependency?

Reply:
I need to check out how, but yes I think yes I can have the setup into a step/dependency